### PR TITLE
chore: fix CreateBlueprintDisplayMetadata function when there is no existing input

### PR DESCRIPTION
**Pull request by tjy9206**
I encountered this error when trying to generate metadata.display.yaml (running cft with `-d` flag):

```
INFO[10-18|17:39:03] Sort variables by the original input order. 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1454fc5]

goroutine 1 [running]:
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/bpmetadata.CreateBlueprintDisplayMetadata({0xc000062c30?, 0x46?}, 0xc000ec4e40, 0xc000680cc0)
        /usr/local/google/home/jieyutian/cloud-foundation-toolkit/cli/bpmetadata/cmd.go:311 +0x365
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/bpmetadata.generateMetadataForBpPath({0xc000062c30, 0x46})
        /usr/local/google/home/jieyutian/cloud-foundation-toolkit/cli/bpmetadata/cmd.go:179 +0x205
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/bpmetadata.generate(0x49ffa00?, {0xc000680c60?, 0x4?, 0x2b7fbf1?})
        /usr/local/google/home/jieyutian/cloud-foundation-toolkit/cli/bpmetadata/cmd.go:125 +0x607
github.com/spf13/cobra.(*Command).execute(0x49ffa00, {0xc000680c00, 0x6, 0x6})
        /usr/local/google/home/jieyutian/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0x49fe8c0)
        /usr/local/google/home/jieyutian/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /usr/local/google/home/jieyutian/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/cmd.Execute()
        /usr/local/google/home/jieyutian/cloud-foundation-toolkit/cli/cmd/root.go:86 +0x1a
main.main()
        /usr/local/google/home/jieyutian/cloud-foundation-toolkit/cli/main.go:6 +0xf
```

This is because we are not checking if `bpCore.Spec.Ui.Input` has any null pointer references in `existingInput := proto.Clone(bpCore.Spec.Ui.Input).(*BlueprintUIInput)`. Fixing in this PR.

### CODE REVIEW

**Review by qz267 — APPROVED** [review]
LGTM, thanks @tjy9206

File: cli/bpmetadata/cmd.go [hunk 1]
```diff
@@ -308,7 +308,12 @@ func CreateBlueprintDisplayMetadata(bpPath string, bpDisp, bpCore *BlueprintMeta
 	bpDisp.Spec.Info.Source = bpCore.Spec.Info.Source
 	buildUIInputFromVariables(bpCore.Spec.Interfaces.Variables, bpDisp.Spec.Ui.Input)
 
-	existingInput := proto.Clone(bpCore.Spec.Ui.Input).(*BlueprintUIInput)
+	existingInput := func() *BlueprintUIInput {
```
Comment (bharathkkb) [inline cli/bpmetadata/cmd.go:311]: Could we add a test for this case too?

**Review by bharathkkb — APPROVED** [review]

File: cli/bpmetadata/cmd.go (hunk 1 above)
Comment (qz267) [reply cli/bpmetadata/cmd.go:311]: let me add one in follow up PR.

Comment (tjy9206) [reply cli/bpmetadata/cmd.go:311]: This is in cmd.go which I don't think we have unit testing coverage. Do we want to create unit tests for this file? Also @qz267 do we already have some other tests for the display metadata generation?

Comment (tjy9206) [reply cli/bpmetadata/cmd.go:311]: I would prefer we merge this first and do the refactoring later so we can also try extract other logics from cmd.go that need unit testing. Created a bug for this.

Comment (tjy9206) [reply cli/bpmetadata/cmd.go:311]: @qz267 shall we include this as part of our testing improvement plan?

**Pull request merged**